armi/reactor/reactors.py

```diff
@@ -2321,3 +2327,30 @@ def processLoading(self, cs, dbLoad: bool = False):
         self.p.maxAssemNum = self.getMaxParam("assemNum")
 
         getPluginManagerOrFail().hook.onProcessCoreLoading(core=self, cs=cs)
+
+    def _updateBlockBOLHeights(self, assems: list, dbLoad: bool, detAxExp: bool):
+        """expand assemblies, resolve disjoint axial mesh (if needed), and update block BOL heights"""
+        axialExpChngr = AxialExpansionChanger(detAxExp)
+        for a in assems:
+            if not a.hasFlags(Flags.CONTROL):
+                axialExpChngr.setAssembly(a)
+                axialExpChngr.expansionData.computeThermalExpansionFactors()
+                axialExpChngr.axiallyExpandAssembly(thermal=True)
+        # resolve axially disjoint mesh (if needed)
+        if not dbLoad:
+            axialExpChngr.manageCoreMesh(self.parent)
+            self.p.referenceBlockAxialMesh = self.findAllAxialMeshPoints(
+                applySubMesh=False
+            )
+            self.p.axialMesh = self.findAllAxialMeshPoints()
+        else:
+            if not detAxExp:
+                # calculate refMesh once for efficiency
+                refMesh = self.refAssem.getAxialMesh()
```

Comment:
This needs to be the finestMeshAssembly. We don't want to expand blueprints to the post burnup heights. Blueprints should be loaded the same with the same mesh regardless of the time node.

Reply:
Addressed https://github.com/terrapower/armi/pull/823/commits/76cfc4f4095a94773a16c5f0b808838db7f36d4c